p2p/src/interface/p2p_interface.rs

```diff
@@ -25,6 +25,10 @@ pub trait P2pInterface: Send + Sync {
     async fn connect(&mut self, addr: String) -> crate::Result<()>;
     async fn disconnect(&mut self, peer_id: PeerId) -> crate::Result<()>;
 
+    async fn list_banned(&mut self) -> crate::Result<Vec<String>>;
```

Comment:
Could we do better than `Vec<String>` here? String manipulation tends to be brittle and exposes whole bunch of operations that are nonsensical for ban address identifiers.

Reply:
We could, but we would have to remove generic addresses from `NetworkingService` and instead use concrete types  (`SocketAddress`/`IpAddress`):
```
pub trait NetworkingService {
    type Address...
    type BannableAddress...
```
It's probably not a good time to do this, and I don't know how else to do it.

Reply:
Ok, don't worry about it now